mruby/src/value/mod.rs

```diff
@@ -75,6 +80,32 @@ impl Value {
     }
 }
 
+impl Clone for Value {
+    fn clone(&self) -> Self {
+        let ruby_type = self.ruby_type();
+        if ruby_type == Data {
+            unsafe {
+                let ptr = mem::transmute::<*mut c_void, Rc<RefCell<_>>>(self.value.value.p);
+                let cloned_ptr = Rc::clone(&ptr);
```

Comment:
after we clone `ptr`, we need to call `mem::forget(ptr)`, otherwise when `ptr` goes out of scope, it will be dropped which will cause the `void *` in `self` to point to garbage. This is a similar idea to this code:

https://github.com/lopopolo/ferrocarril/blob/9c81922cb87725d6284585d00c7cd1bf4859978c/mruby/src/interpreter.rs#L142-L152